### DISCUSSION TITLE
Global app data from CMS

### DISCUSSION THREAD

**Discussion by tobiastornros** [vercel/next.js#12595] [Help] [answered]
Hi,

We use Sanity as CMS for our sites and we have a page configured in the CMS to collect all information that are "static"/global for the site. Often called Site Settings in the CMS-world. It could be information or content such as social icons, images, company information for footer, cookie banner content for example.

Where are we supposed to fetch this information on first render/build time with next.js?

**Comment by rafaelalmeidatk** [comment, marked as answer]
The [`_app`](https://nextjs.org/docs/advanced-features/custom-app) page seems to be a good place to fetch this info, since it is a wrapper around all the pages

**Reply by scopsy** [reply]
Does using a query there disables SSG?

**Reply by rafaelalmeidatk** [reply]
If you add `getInitialProps` to `_app` it will disable [Automatic Static Optimization](https://nextjs.org/docs/advanced-features/automatic-static-optimization), but you can still have static pages by adding `getStaticProps` to each page

**Reply by TommySorensen** [reply]
@rafaelalmeidatk Yes, as the documentation mention:
Only uncomment this method if you have blocking data requirements for every single page in your application.


https://nextjs.org/docs/advanced-features/custom-app
```js
// import App from 'next/app'

function MyApp({ Component, pageProps }) {
  return <Component {...pageProps} />
}

// Only uncomment this method if you have blocking data requirements for
// every single page in your application. This disables the ability to
// perform automatic static optimization, causing every page in your app to
// be server-side rendered.
//
// MyApp.getInitialProps = async (appContext) => {
//   // calls page's `getInitialProps` and fills `appProps.pageProps`
//   const appProps = await App.getInitialProps(appContext);
//
//   return { ...appProps }
// }

export default MyApp
```

@tobiastornros  I will suggest you just fetch the data on each page with `getStaticProps` and then add the new [revalidation](https://nextjs.org/blog/next-9-4#incremental-static-regeneration-beta) to update pages after some seconds
